src/Connection/DefaultConnectionPool.php

```diff
@@ -132,6 +133,7 @@ public function getConnection(Request $request, CancellationToken $cancellation)
                     return;
                 }
 
+                // Replace Promise with Connection object.
```

Comment:
Why not just keep the `Promise` and always do `$connection = yield $connection`?

Reply:
That would be simpler. Will change.